src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt

```diff
@@ -237,4 +237,19 @@ class BookingControllerMockMvcTests @Autowired constructor(
         mvc.perform(delete("/v1/booking/999"))
             .andExpect(status().isNoContent)
     }
+
+    @TestConfiguration
+    class InternalConfig {
+        @Bean
+        fun configurer(): WebMvcConfigurer = object : WebMvcConfigurerAdapter() {
+            override fun addFormatters(registry: FormatterRegistry) {
+                registry.addConverter(String::class.java, Booking::class.java) { id ->
+                    when (id) {
+                        BookingControllerMockMvcTests.booking.id -> BookingControllerMockMvcTests.booking
+                        else -> null
```

Comment:
this magic is needed to handle the id -> object "lookup" that spring does for us now... almost decided this was too "magic" and went back to what we had... thoughts?

Reply:
Witchcraft.  Burn it.

Just kidding!  Too disruptive to change back away from it based on the work you've done here, IMO.  But my brain still recoils to `getFoo(Foo)`.